src/main/java/org/apache/maven/plugin/surefire/report/TreePrinter.java

```diff
@@ -155,7 +157,8 @@ private void printClass() {
             } else {
                 builder.a(theme.entry());
             }
-            println(concatenateWithTestGroup(builder, testResult, !isBlank(testResult.getReportNameWithGroup())));
+            String message = concatenateWithTestGroup(builder, testResult, !isBlank(testResult.getReportNameWithGroup()));
+            println(message + " - " + classResult.elapsedTimeAsString() + "s");
```

Comment:
Now it's printing like this:
```
[INFO] +--Nested Sample - 0.112s
[INFO] |  +-- [OK] Should pass - 0.001s
[INFO] |  '-- [OK] Should pass2 - 0.001s
[INFO] +--.--Inner Test - 0.112s
[INFO] |  |  '-- [OK] Inner test should pass - 0.001s
[INFO] |  '--.--Inner Inner Test - 0.112s
[INFO] |     |  '-- [OK] Inner Inner Test should pass - 0.001s
[INFO] |     '-----Inner Inner Inner Test - 0.112s
[INFO] |           '-- [OK] Inner Inner Inner Test should pass - 0.001s
```
It would be better to check for the treeLength and print the timing only for the root class, since we cannot check for all inner class timings and it's printing an inaccurate data.